mono/metadata/threads.c

```diff
@@ -4849,9 +4849,9 @@ self_abort_internal (void)
 
 	exc = mono_thread_request_interruption (TRUE);
 	if (exc)
-		mono_raise_exception (exc);
-
-	mono_thread_info_self_interrupt ();
+		mono_set_pending_exception (exc);
```

Comment:
why not use a MonoError with `mono_error_set_exception_instance` and in the caller to `self_abort_internal` have a `mono_error_set_pending_exception` ?